tests/specs/neogit/operations_spec.lua

```diff
@@ -46,4 +46,24 @@ describe("branch popup", function()
       eq("branch-from-test", get_current_branch())
     end)
   )
+
+  it(
```

Comment:
These tests aren't complete - what is the best way to setup up commits and verify which commits are present?

Reply:
That aspect of our test suite is still a bit of a work in progress, and @treatybreaker  might have some ideas here, but you can just use util.system to interact with the normal git binary, make a commit, and read the revision sha and/or the HEAD file. I can probably explain better when I'm not on my phone, if thats not enough to go off ☺️

Reply:
Yeah, best way to go for now is to liberally abuse `util.system`.

To set up commits what I recommend is "cleaning" the repository by stashing and clearing that stash like so: 

```lua
local util = require("tests.util.util")

util.system("git stash && git stash clear")
```

Then creating the commits you need like so:

```lua
local util = require("tests.util.util")

util.system("touch random-file && git commit -m 'Commit Message'")
```

This will guarantee that the most recent commit is "Commit Message".

The other thing to note is that the original commits and test repository state can be found by going into `tests/.repo` then changing the `.git.orig` file in there to `.git` and that will make the base test repository we use accessible via the git cli.

I need get around to doing a update to the test utilities so we have some desired state configuration for the repository setup entirely in Lua so this type of stuff is less of a PIA.

Reply:
@morgsmccauley I just merged a PR which adds `exec` inside `git_harness.lua` for testing.

It allows you to run an arbitrary git command `{ "git", "log", "--one-line" }` etc as a list of arguments, and it returns a list of the lines.

I think that will greatly help

Reply:
@ten3roberts it looks like `vim.system()` is only available in the release build of Neovim, so this will fail in the [test action](https://github.com/NeogitOrg/neogit/blob/master/.github/workflows/test.yml#L31C53-L31C53) which uses the stable build.

Is there another way to achieve this or should we target release in the action?

Reply:
Actually I didn't need to use `exec`, some of the other existing functions were enough.

I've added tests for both spin out/off now :)